app/controllers/fhir/r4/api_controller.rb

```diff
@@ -210,6 +210,14 @@ def create
     when 'patient'
       return if doorkeeper_authorize!(*PATIENT_WRITE_SCOPES)
 
+      # Check for duplicates
+      if request.headers['If-None-Exist'].present?
+        search_params = request.headers['If-None-Exist'].split('&').map { |p| p.split('=', 2) }.select { |p| p.size == 2 }.to_h
+        num_matches = search_patients(search_params).size
+        err_msg = "There #{num_matches > 1 ? 'are' : 'is'} #{num_matches} potential duplicate patient#{num_matches > 1 ? 's' : ''}"
```

Comment:
In the case of 1 duplicate, I believe the spec indicates that we are supposed to just return that duplicate with a `200 OK` (as opposed to `201 created`). Reference [here](https://www.hl7.org/fhir/http.html#ccreate)

Reply:
I'm not sure that the duplicate needs to be returned, just a 200 based on the reference

Reply:
Instead of building a string like this take a look at https://stackoverflow.com/questions/8790093/how-to-pluralize-there-is-are-n-object-objects and https://api.rubyonrails.org/classes/ActionView/Helpers/TextHelper.html#method-i-pluralize

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1098/commits/b335e9bae20603fd40a4b1f465c206aff9f4b9ef#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R282

Reply:
@Bialogs as for your 2nd comment, that logic is no longer relevant because different logic is applied depending on whether 1 or multiple results are found